Frameworks/AEPSwiftUI/UnitTests/AEPStackTests.swift

```diff
@@ -0,0 +1,164 @@
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import XCTest
+import SwiftUI
+@testable import AEPSwiftUI
+
+final class AEPStackTests: XCTestCase {
+
+    let mockModel = AEPText([Constants.CardTemplate.UIElement.Text.CONTENT: "Text Content"])!
+
+    func testHStack_Init() {
+        // test
+        let hStack = AEPHStack()
+
+        // verify
+        XCTAssertNotNil(hStack)
+        XCTAssertNotNil(hStack.view)
+    }
+
+    func testHStack_addModel() {
+        // test
+        let hStack = AEPHStack()
+        hStack.addModel(mockModel)
+
+        // verify
+        XCTAssertEqual(hStack.childModels.count, 1)
+    }
+
+    func testHStack_addView() {
+        // test
+        let hStack = AEPHStack()
+        hStack.addView(Text("Hello"))
+
+        // verify
+        XCTAssertEqual(hStack.childModels.count, 1)
+
+        // test again
+        hStack.addView(Text("It's me"))
+
+        // verify
+        XCTAssertEqual(hStack.childModels.count, 2)
+        XCTAssertNotNil(hStack.view)
+    }
+
+    func testHStack_removeView() {
+        // setup
+        let hStack = AEPHStack()
+        hStack.addView(Text("Hello"))
+        hStack.addView(Text("Hello Again"))
+
+        // verify
+        XCTAssertEqual(hStack.childModels.count, 2)
+
+        // test and verify
+        hStack.removeView(at: 1)
+        XCTAssertEqual(hStack.childModels.count, 1)
```

Comment:
can we verify that the correct view is also removed here?

Reply:
It is not possible at the moment.
XCTest does not have the capabilities evaluate or inspect SwiftUI Views. 
We need to explore other tools like https://github.com/nalexn/ViewInspector to get the job done.

In this scenario, I can even evaluate the model class.. But it needs an ID to identify itself. 
I will add ID to all model classes in my next PR